src/Microsoft.Health.Fhir.SqlServer/Features/Search/SqlServerSearchService.cs

```diff
@@ -248,7 +252,7 @@ private async Task<SearchResult> SearchImpl(SearchOptions searchOptions, bool hi
 
                             // At this point we are at the last row.
                             // if we have more columns, it means sort expressions were added.
-                            if (reader.FieldCount > 10)
+                            if (reader.FieldCount > _resourceTableColumnCount)
```

Comment:
>_resourceTableColumnCount [](start = 52, length = 25)

rant: If you will ever have chance to touch sql query generator  I think this should be static property of class. (VLatest.Resource.ColumnCount)
It's too easy to forget to update this field in the future.

(As option you can calculate via reflection how many properties of class derived from `Column` base class, but I think it's too much)

Reply:
Good idea. I originally attempted to implement reflection to pull in the number of properties, but, as you said, it was a little over the top. I created the work item [AB#79261](https://microsofthealth.visualstudio.com/Health/_workitems/edit/79261) to track this.